### PR TITLE
Turtlelord26 Melee 2.1 Melee Weapon Categorization

### PR DESCRIPTION
**Pull request by Turtlelord26**
Solves Issue #411 

I have added 5 Size categories of Melee Weapon: Light, One-handed, Two-handed, Polearm, and Exotic. The first four work like gun categories for proficiency purposes, Exotic is more special and applies to categories like "whip" or "flail". 
Also took a first pass on giving the various classes baseline melee proficiencies. Some class passives were scope-creeped on during this process, see commit notes. A couple of explains:
-Assassin only gets the smaller ones because that's their aesthetic
-Hunters get polearms because hunting and one-handed because machetes.
Finally, there are feats now for expanding individual characters' melee repertoires.

### CODE REVIEW

File: CharacterCreation/03_Feats.txt [hunk 1]
```diff
@@ -402,6 +402,51 @@ Prerequisites:
...
+
+	Allows a character to use Melee weapons of size Light.
+
+== Proficiency: Polearm Melee Weaponry ==
+
+Prerequisites: 
+* Proficiency: Light Melee Weaponry
+
+	Allows a character to use Melee weapons of size Polearm.
+
```
Comment (bleehu) [inline CharacterCreation/03_Feats.txt]: ... polearms? Seems a little... out of aesthetic for a squad shooter game?

**Review by bleehu — APPROVED** [review]
I'm a little concerned about the aethetic of polearms in space? But that's non-blocking as far as I'm concerned. Shipit!

**Comment by Turtlelord26** [comment]
I thought I would support Polearms specifically for the Hunter aesthetic. Spears are handy for hunting the larger beasties if you don't have a .70 caliber shotgun. That said, I am open to discussing them.

**Comment by Turtlelord26** [comment]
After further reflection, I realized I wouldn't mind reclassifying the polearms as Exotic - Two-handed. Will update.

**Comment by bleehu** [comment]
Shipit flavored and drizzled in shipit sauce.

**Comment by Turtlelord26** [comment]
That is a comment that was submitted at lunch time. 👍

**Comment by bleehu** [comment]
coincidence

File: CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt [hunk 2]
```diff
@@ -105,11 +135,12 @@ DEX Requirement: 6
...
 Damage: 50 + 6*STR
+Size: Two-handed
 Range: 2m
 Accuracy Modifier: +0
 STR Requirement: 6
 
-	2-handed weapon. Considered concussive (explosive) damage. Hits armor points 
+	Considered concussive (explosive) damage. Hits armor points 
 before physical damage if the target's armor has defense against explosive (i.e. 
 concussive) damage. If armor roll is greater than 50%, target rolls a DC 50%
```
Comment (rosebrallier) [inline CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt:160]: Are there still armors that provide defense against explosive?
(this is probably scope creep. Ooops.)

File: CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt [hunk 3]
```diff
@@ -8,7 +8,30 @@ and any relevant feat or skill bonuses to oppose your opponent's
...
+and are more commonly acquired by feat.
+
+	When a Two-handed weapon uses a character's STR mod in its damage 
+calculation and the character's STR mod is positive, that character instead 
+uses 150% of their STR modifier to calculate damage.
+
+	Some One-handed weapons have the Hybrid size property. Characters 
+with proficiency with Two-handed melee weapons may make two-handed strikes 
+with these weapons, which allows them to add their STR modifier as if 
+the weapon was Two-handed.
```
Comment (rosebrallier) [inline CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt]: It's a little unclear to me how this works in practice. Maybe add an example?

**Review by rosebrallier — CHANGES_REQUESTED** [review]
The conversion of one- to two-handed dmg for Hybrid weapons is unclear to me and may benefit from an explicit example.

File: CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt (hunk 2 above)
Comment (Turtlelord26) [reply CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt:160]: Yes, definitely scope creep. I was just killing the now-out-of-place reference to being two-handed here.

**Comment by Turtlelord26** [comment]
How's that?

File: CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt [hunk 4]
```diff
@@ -18,12 +18,19 @@ and are more commonly acquired by feat.
 
 	When a Two-handed weapon uses a character's STR mod in its damage 
 calculation and the character's STR mod is positive, that character instead 
-uses 150% of their STR modifier to calculate damage.
+uses 150% of their STR modifier to calculate damage. For example, if some 
+two-handed weapon dealt damage equal to 40 + 2xSTR nod and the character 
+wielding it had a STR mod of -4, each strike would deal 32 = 40 + (-4 * 2) 
+damage. A character with the same weapon and a STR mod of 12 would deal 
+76 = 40 + ((12 * 1.5) * 2) = 40 + (18 * 2) damage per strike.
```
Comment (rosebrallier) [inline CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt:25]: This is a great example, but not what I found confusing.

File: CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt [hunk 5]
```diff
@@ -18,12 +18,19 @@ and are more commonly acquired by feat.
...
 
 	Some One-handed weapons have the Hybrid size property. Characters 
 with proficiency with Two-handed melee weapons may make two-handed strikes 
 with these weapons, which allows them to add their STR modifier as if 
-the weapon was Two-handed.
+the weapon was Two-handed (see the preceeding paragraph). Such a strike 
+must be declared to be two-handed before rolling to hit, and the character 
+doing so loses the benefits of any items that require his or her off hand to 
+function.
```
Comment (rosebrallier) [inline CharacterCreation/04_01_Melee_&_Thrown_Weapons.txt]: What I wanted was and example of how the damage calculation _changes_ when you convert a hybrid from one- to -two-handed

**Review by rosebrallier — CHANGES_REQUESTED** [review]

**Comment by Turtlelord26** [comment]
Tried again

**Review by rosebrallier — APPROVED** [review]

**Comment by rosebrallier** [comment]
Hurrah!
